c2_components/include/mfx_c2_decoder_component.h

```diff
@@ -215,6 +218,11 @@ class MfxC2DecoderComponent : public MfxC2Component
 
     std::vector<std::unique_ptr<C2Param>> m_updatingC2Configures;
 
+    uint64_t m_consumerUsage;
+
+    uint32_t m_surfaceNum;
+    std::list<std::shared_ptr<mfxFrameSurface1>> m_surfacePool; // used in case of system memory
```

Comment:
Maybe better to rename to m_swSurfacePool and m_surfaces rename to m_hwSurfacePool?